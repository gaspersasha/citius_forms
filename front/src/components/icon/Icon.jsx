import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({ type, classes }) => {
  if (!type.length) return null;
  const cls = type.map((t) => `icon--${t}`).join(' ');

  return <span className={`icon ${cls} ${classes}`} />;
};

Icon.propTypes = {
  type: PropTypes.array,
  classes: PropTypes.array,
};
Icon.defaultProps = {
  type: [],
  classes: [],
};

export default Icon;
