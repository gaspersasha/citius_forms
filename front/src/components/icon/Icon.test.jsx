import React from 'react';
import Icon from './Icon';

describe('Icon', () => {
  it("doesn't get rendered without type", () => {
    const wrapper = mount(<Icon />);

    expect(wrapper).toMatchSnapshot();
  });

  it('info icon', () => {
    const wrapper = mount(<Icon type={['info']} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('info alert', () => {
    const wrapper = mount(<Icon type={['exclamation']} />);

    expect(wrapper).toMatchSnapshot();
  });
});
