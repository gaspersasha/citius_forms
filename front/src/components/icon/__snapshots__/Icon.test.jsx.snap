// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon doesn't get rendered without type 1`] = `
<Icon
  classes={Array []}
  type={Array []}
/>
`;

exports[`Icon info alert 1`] = `
<Icon
  classes={Array []}
  type={
    Array [
      "exclamation",
    ]
  }
>
  <span
    className="icon icon--exclamation "
  />
</Icon>
`;

exports[`Icon info icon 1`] = `
<Icon
  classes={Array []}
  type={
    Array [
      "info",
    ]
  }
>
  <span
    className="icon icon--info "
  />
</Icon>
`;
